web/main.py

```diff
@@ -43,17 +62,32 @@ def handler(dataset=None, filename=None):
         logger.warning(f'Invalid dataset "{dataset}"')
         abort(400)
 
+    bucket_name = f'cpg-{dataset}-{BUCKET_SUFFIX}'
+    bucket = storage_client.bucket(bucket_name)
+
     dataset_project_id = dataset_config['projectId']
     members = read_secret(
         dataset_project_id, f'{dataset}-web-access-members-cache'
     ).split(',')
+
     if email not in members:
-        logger.warning(f'{email} is not a member of the {dataset} access group')
-        abort(403)
+        # Second chance: if there's a '.access' file in the first subdirectory,
+        # check if the email is listed there.
+        split_subdir = filename.split('/', maxsplit=1)
+        if len(split_subdir) == 2 and split_subdir[0]:
+            access_list_filename = f'{split_subdir[0]}/.access'
```

Comment:
```suggestion
            access_list_filename = os.path.join(subdir, '.access')
```

Reply:
At the moment, `split_subdir[0]` can't contain any slashes, so `os.path.join` doesn't really help. Would be cleaner if we used `dirname` as you suggested, but that would have different semantics.